chalice/app.py

```diff
@@ -266,15 +274,29 @@ def __init__(self, body, headers=None, status_code=200):
         self.headers = headers
         self.status_code = status_code
 
-    def to_dict(self):
+    def _base64encode(self, data):
+        if not isinstance(data, bytes):
+            raise ValueError('Expected bytes type for body with binary '
+                             'Content-Type. Got %s type body instead.'
+                             % type(data))
+        data = base64.b64encode(data)
+        return data.decode('ascii')
+
+    def to_dict(self, binary_types=None):
+        if binary_types is None:
+            binary_types = []
         body = self.body
-        if not isinstance(body, str):
-            body = json.dumps(body, default=handle_decimals)
-        return {
+        response = {
             'headers': self.headers,
             'statusCode': self.status_code,
-            'body': body,
         }
+        if self.headers.get('Content-Type') in binary_types:
+            body = self._base64encode(body)
+            response['isBase64Encoded'] = True
+        elif not isinstance(body, str):
```

Comment:
It's admittedly an edge case, but shouldn't we allow a user to mark `application/json` as a binary type and yet still return a dict/list, etc:

```
app.api.whatever_the_api_is(['application/json'])

@app.route('/')
def foo():
   return {"this will be json": "bar"}
```

Weird use case, but it seems odd that we should start breaking if they happen to mark that as a binary type.